src/js/iov/MSEWrapper.js

```diff
@@ -642,6 +694,57 @@ export default class MSEWrapper {
     this.metrics = null;
     this.events = null;
     this.eventListeners = null;
-    this.onSourceBufferUpdateEnd = null;
+
+    debug('_freeAllResources finished...');
+  }
+
+  destroy () {
+    debug('destroy...');
+
+    if (this.destroyed) {
+      return Promise.resolve();
+    }
+
+    this.destroyed = true;
+
+    this.destroyMediaSource();
+
+    // We MUST not force the destroy method here to be asynchronous, even
+    // though it "should" be.  This is because we cannot assume that the
+    // caller has control over whether or not its destroy method can be
+    // properly run asynchronously.  The specific use case here is that
+    // many client side libraries (angular, marionette, react, etc.) do
+    // not all give pre-destruction methods or events that can wait for
+    // an asynchronous operation.  If angular decides it is going to
+    // destroy a DOM element when a user navigates, we have no way of
+    // ensuring that it supports asynchronous operations, or that the
+    // caller is properly using them, if they exist.  Therefore, this
+    // destroy method will clean up the source buffer later, allowing the
+    // rest of the clsp destruction logic to continue.  The use case for
+    // needing that functionality is that the conduit needs to use the its
+    // iframe to contact the server, and if the iframe is destroyed before
+    // the conduit talks to the server, errors will be thrown during
+    // destruction, which will lead to resources not being free / memory
+    // leaks, which may cause the browser to crash after extended periods
+    // of time, such as 24 hours.
+    // Note that we still return the promise, so that the caller has the
+    // option of waiting if they choose.
+    const destroyPromise = this.destroySourceBuffer()
+      .then(() => {
+        debug('destroySourceBuffer successfully finished...');
+        this._freeAllResources();
+      })
+      .catch((error) => {
+        debug('destroySourceBuffer failed...');
+        console.error('Error while destroying the source buffer!');
+        console.error(error);
+
+        // Do our best at memory management, even on failure
+        this._freeAllResources();
+      });
+
+    debug('destroy finished...');
```

Comment:
This should be 'exiting destroy method' vice 'destroy finished' as the destruction is not finished at this point. Maybe move this up into the then block.